examples/incremental-ingest-from-db-example/10_increm_ingest_from_db_example.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pathlib
+import sqlite3
+
+from vdk.api.job_input import IJobInput
+
+
+def run(job_input: IJobInput):
+
+    # Get last_date property/parameter:
+    #  - if the target table already exists, take the property value already stored in the DJ from the previous run
+    #  - if the target table does not exist, set last_date to 01-01-1900 in oder to fetch all rows
+    last_date = job_input.get_property("last_date", "01-01-1900")
+
+    # Connect to sqlite local db
+    os.chdir(pathlib.Path(__file__).parent.absolute())
+    with sqlite3.connect("data/incremental_ingest_example.db") as db_connection:
```

Comment:
you can set in config.ini SQLITE_FILE (https://github.com/vmware/versatile-data-kit/tree/main/projects/vdk-plugins/vdk-sqlite)

Reply:
I tried the suggested approach, but the `job_input.execute_query` method returns only a list of the rows in the DB and not the column names:
```
[(1, 'record one', '10-01-2021'), (2, 'second record', '10-02-2021'), (3, 'this is record 3', '10-03-2021')]
<class 'list'>
```
For the `job_input.send_tabular_data_for_ingestion()`, however, I need to specify `column_names`. I couldn't find how I can extract the column names using `job_input.execute_query` method anywhere in the documentation or examples. Is there a way to do it?

Reply:
job_input.execute_query() is an out-of-the-box `cursor.execute()`, where the `connect()` has been performed automatically using the VDK configurations. For example, https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/job_input.py#L75
For how to use in the context of sqlite database, see https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-sqlite/README.md

Reply:
Hi Iva, got that, but the issue of using `job_input.execute_query()` in this context is that I cannot retrieve the names of the columns in the sqlite database table, the method returns only the rows in the table as a list object (as shown in the example in my previous comment). I need the column names because after reading the data, I call the `job_input.send_tabular_data_for_ingestion()` where column_names is a required argument. 
If there is no way to get the column names using `job_input.execute_query()`, I think we should keep the current configuration with `sqlite3.connect()` and `cursor.execute()`.

Reply:
Hi, how do you use `cursor.execute()` to retrieve column names? `job_input.execute_query()` would be an alias, so the approach would be the same - inly without the "custom" db connection made

Reply:
This is how it's done currently (part of the current pull request):
```
# Connect to sqlite local db
os.chdir(pathlib.Path(__file__).parent.absolute())
with sqlite3.connect('data/incremental_ingest_example.db') as db_connection:

    # Create a cursor object
    cursor = db_connection.cursor()

    # Select the needed records from the source table using a sqlite query parameter
    cursor.execute(
        f"""
        SELECT * FROM increm_ingest
        WHERE reported_date > '{last_date}'
        ORDER BY reported_date
        """
    )
    data = cursor.fetchall()

    # If any data is returned from the query, send the fetched records for ingestion
    if len(data) > 0:
        job_input.send_tabular_data_for_ingestion(
            data,
            column_names=[column_info[0] for column_info in data.description],
            destination_table="incremental_ingest_from_db_example"
        )
```
Column names are fetched in the following way: `[column_info[0] for column_info in data.description]`. If I try to do the same on the result of `job_input.execute_query()`, I receive an error message that list objects do not have a property `description`.

Reply:
@dvalkova I got it,
one possibility is to use `PRAGMA table_info(increm_ingest)`, then parse the content

Reply:
Thanks Iva, this worked! I will update the code and readme and will commit.